python/ray/tune/search/basic_variant.py

```diff
@@ -360,6 +388,46 @@ def add_configurations(
             self._iterators.append(iterator)
             self._trial_generator = itertools.chain(self._trial_generator, iterator)
 
+    def update_config_reference_table(
+        self, experiments: Union["Experiment", List["Experiment"], Dict[str, Dict]]
+    ):
+        """Update references contained by experiment specs.
+
+        Arguments:
+            experiments: Experiments whose configured specs need to get updated.
+        """
+        for exp in experiments:
+            for k, v in _unresolved_values(exp.spec["config"]).items():
+                if k in self._config_reference_table[exp.name]:
+                    # Update reference values.
+                    self._config_reference_table[exp.name][k] = v
+
+    def _points_to_evaluate_to_indices(self, experiment_name, points: Dict):
+        """Turn categorical values in points_to_evaluate into indices.
+        """
+        for k, v in self._config_reference_table[experiment_name].items():
+            try:
+                values = _get_value(points, k)
+                if isinstance(values, list):
+                    indices = [v.index_of(val) for val in values]
+                else:
+                    indices = v.index_of(values)
+                assign_value(points, k, indices)
+            except KeyError:
+                # It's ok if a categorical value is not specified in
```

Comment:
If the points to evaluate contains something that isn't in the list of Categorical options, then we may fail on restore. Since it doesn't get resolved from the reference table.

This may be an extreme edge case, so one possibility is just to warn the user.

Reply:
n/a now.